File: backend/app/services/ferramentas/cliente.py
from sqlalchemy.orm import Session
from app.models import Cliente


SCHEMA_BUSCAR_CLIENTE = {
    "name": "buscar_cliente",
    "description": "Busca dados de um cliente por nome ou CPF/CNPJ. Retorna dados pessoais, contato e endereco.",
    "input_schema": {
        "type": "object",
        "properties": {
            "busca": {
                "type": "string",
                "description": "Nome parcial ou CPF/CNPJ do cliente",
            },
        },
        "required": ["busca"],
    },
}


def executar_buscar_cliente(input_data: dict, db: Session) -> str:
    busca = input_data.get("busca", "").strip()
    if not busca:
        return "Parametro de busca vazio."

    cliente = db.query(Cliente).filter(Cliente.cpf_cnpj == busca).first()
    if not cliente:
        clientes = (
            db.query(Cliente)
            .filter(Cliente.nome.ilike(f"%{busca}%"))
            .limit(5)
            .all()
        )
        if not clientes:
            return f"Nenhum cliente encontrado para '{busca}'."
        if len(clientes) == 1:
            cliente = clientes[0]
        else:
            linhas = [f"Encontrados {len(clientes)} clientes:"]
            for c in clientes:
                linhas.append(f"  ID {c.id}: {c.nome} (CPF/CNPJ: {c.cpf_cnpj})")
            return "\n".join(linhas)

    linhas = [
        f"CLIENTE: {cliente.nome}",
        f"CPF/CNPJ: {cliente.cpf_cnpj}",
        f"Telefone: {cliente.telefone}",
    ]
    if cliente.email:
        linhas.append(f"Email: {cliente.email}")
    if cliente.endereco:
        linhas.append(f"Endereco: {cliente.endereco}")
        if cliente.cidade:
            linhas.append(f"Cidade: {cliente.cidade}/{cliente.uf} CEP: {cliente.cep or 'N/A'}")
    if cliente.profissao:
        linhas.append(f"Profissao: {cliente.profissao}")
    if cliente.estado_civil:
        linhas.append(f"Estado Civil: {cliente.estado_civil}")

    return "\n".join(linhas)

File: frontend/lib/models/cliente.dart
class Cliente {
  final int id;
  final String nome;
  final String cpfCnpj;
  final String? rg;
  final String? cnh;
  final String? dataNascimento;
  final String? nacionalidade;
  final String? estadoCivil;
  final String? profissao;
  final String telefone;
  final String? telefone2;
  final String? email;
  final String? endereco;
  final String? cidade;
  final String? uf;
  final String? cep;
  final String? observacoes;
  final String? outrosDados;
  final DateTime createdAt;
  final DateTime updatedAt;

  Cliente({
    required this.id,
    required this.nome,
    required this.cpfCnpj,
    this.rg,
    this.cnh,
    this.dataNascimento,
    this.nacionalidade,
    this.estadoCivil,
    this.profissao,
    required this.telefone,
    this.telefone2,
    this.email,
    this.endereco,
    this.cidade,
    this.uf,
    this.cep,
    this.observacoes,
    this.outrosDados,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Cliente.fromJson(Map<String, dynamic> json) {
    return Cliente(
      id: json['id'],
      nome: json['nome'],
      cpfCnpj: json['cpf_cnpj'],
      rg: json['rg'],
      cnh: json['cnh'],
      dataNascimento: json['data_nascimento'],
      nacionalidade: json['nacionalidade'],
      estadoCivil: json['estado_civil'],
      profissao: json['profissao'],
      telefone: json['telefone'],
      telefone2: json['telefone2'],
      email: json['email'],
      endereco: json['endereco'],
      cidade: json['cidade'],
      uf: json['uf'],
      cep: json['cep'],
      observacoes: json['observacoes'],
      outrosDados: json['outros_dados'],
      createdAt: DateTime.parse(json['created_at']),
      updatedAt: DateTime.parse(json['updated_at']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nome': nome,
      'cpf_cnpj': cpfCnpj,
      'rg': rg,
      'cnh': cnh,
      'data_nascimento': dataNascimento,
      'nacionalidade': nacionalidade,
      'estado_civil': estadoCivil,
      'profissao': profissao,
      'telefone': telefone,
      'telefone2': telefone2,
      'email': email,
      'endereco': endereco,
      'cidade': cidade,
      'uf': uf,
      'cep': cep,
      'observacoes': observacoes,
      'outros_dados': outrosDados,
    };
  }
}
